front/lib/api/assistant/participants.ts

```diff
@@ -0,0 +1,148 @@
+import type {
+  LightAgentConfigurationType,
+  ModelId,
+  Result,
+} from "@dust-tt/types";
+import { Err, formatUserFullName, Ok } from "@dust-tt/types";
+import { Op } from "sequelize";
+
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import type { Authenticator } from "@app/lib/auth";
+import {
+  AgentMessage,
+  Conversation,
+  Message,
+  User,
+  UserMessage,
+} from "@app/lib/models";
+
+interface AgentParticipant {
+  configurationId: string;
+  name: string;
+  pictureUrl: string;
+}
+
+interface UserParticipant {
+  fullName: string | null;
+  pictureUrl: string | null;
+  username: string;
+}
+
+export interface FetchConversationParticipantsResponse {
+  participants: {
+    agents: AgentParticipant[];
+    users: UserParticipant[];
+  };
+}
+
+async function fetchAllUsersById(
+  userIds: ModelId[]
+): Promise<UserParticipant[]> {
+  const users = (
+    await User.findAll({
+      attributes: ["firstName", "lastName", "imageUrl", "username"],
+      where: {
+        id: {
+          [Op.in]: userIds,
+        },
+      },
+    })
+  ).filter((u) => u !== null) as User[];
+
+  return users.map((u) => ({
+    fullName: formatUserFullName(u),
+    pictureUrl: u.imageUrl,
+    username: u.username,
+  }));
+}
+
+async function fetchAllAgentsById(
+  auth: Authenticator,
+  agentConfigurationIds: string[]
+): Promise<AgentParticipant[]> {
+  // TODO(2024-3-25 flav) Support fetching many agents by id.
+  const agents = (
+    await Promise.all(
+      agentConfigurationIds.map((agentConfigId) => {
+        return getAgentConfiguration(auth, agentConfigId);
+      })
+    )
+  ).filter((a) => a !== null) as LightAgentConfigurationType[];
+
+  return agents.map((a) => ({
+    configurationId: a.sId,
+    name: a.name,
+    pictureUrl: a.pictureUrl,
+  }));
+}
+
+export async function fetchConversationParticipants(
+  auth: Authenticator,
+  conversationId: string
```

Comment:
Let's pass `ConverstationWithoutContentType` since we have it ?

Reply:
Good catch!